pyzoo/zoo/orca/data/ray_xshards.py

```
@@ -331,6 +331,9 @@ def _from_spark_xshards_ray_api(spark_xshards):
                 # if running in cluster, filter out driver ip
                 if ray_ctx.is_local or key != f"node:{driver_ip}":
                     nodes.append(key)
+        # for the case of single node spark standalone
+        if not nodes:
+            nodes.append(f"node:{driver_ip}")
```

Also check node:{driver_ip} in keys? Do we need to raise an error if it is not in keys and nodes=None?

---

Ray may make sure at least one node in resources, so nodes will not be empty. In that case, maybe we don't need to raise error?

---

If node:{driver_ip} isn't in keys, it should be added to nodes in the condition of `key != f"node:{driver_ip}`

---

ok Then we don't need to raise error.